### PR TITLE
os/bluestore: introduce health alert on spurious read errors.

### PR DESCRIPTION
**Pull request by ifed01**
Signed-off-by: Igor Fedotov <ifedotov@suse.com>

## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard backend`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

**Comment by ifed01** [comment]
jenkins make docs please

**Comment by ifed01** [comment]
jenkins render docs

**Comment by ifed01** [comment]
jenkins retest this please

**Comment by ifed01** [comment]
jenkins retest this please

File: src/os/bluestore/BlueStore.h [hunk 1]
```diff
@@ -2916,11 +2916,12 @@ class BlueStore : public ObjectStore,
 private:
   ceph::mutex qlock = ceph::make_mutex("BlueStore::Alerts::qlock");
   std::string failed_cmode;
-  std::set<std::string> failed_compressors;
```
Comment (jdurgin) [inline src/os/bluestore/BlueStore.h:2919]: nit: unrelated removal of std::

**Comment by badone** [comment]
[wip-badone-testing](https://trello.com/c/efwUoBaI/867-wip-badone-testing)

**Comment by badone** [comment]
http://pulpito.ceph.com/bhubbard-2020-04-16_09:57:54-rados-wip-badone-testing-distro-basic-smithi/

Tracking down failures but I don't think any are related.

**Pull request merged**